### ISSUE TITLE
I think let and const will actually be hoisted in their scope

### ISSUE THREAD

**Issue by DanielWLam** [DrkSephy/es6-cheatsheet#86]
when you declare a variable with var, it will be hoisted and set value as undefined.
I think let and const are so similar but they are passed the default value 'ReferenceError'

**Comment by DrkSephy** [comment]
@DanielWLam you are correct. `let` and `const` declarations are hoisted, however if you try to reference these variables before they are declared, a `ReferenceError` is thrown (regardless of the hoisting. This is also known as the [Temporal Dead Zone](https://ponyfoo.com/articles/es6-let-const-and-temporal-dead-zone-in-depth).
